hotspot: 1.4.1 -> 1.5.0

K900 opened the pull request:
## Description of changes

Finally!

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

K900 committed:
kddockwidgets: 1.7.0 -> 2.0.0

K900 committed:
hotspot: 1.4.1 -> 1.5.0

Staying on Qt5 for now, because Qt6 is missing features

SuperSandro2000 commented on pkgs/development/tools/analysis/hotspot/default.nix:
```diff
@@ -64,6 +70,7 @@ mkDerivation rec {
 
   qtWrapperArgs = [
     "--prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ rustc-demangle ]}"
```
Do we still need LD_LIBRARY_PATH here?

K900 replied:
It does dlopen rustc_demangle. We could probably replace it with a patchelf call.

alois31 commented on pkgs/development/tools/analysis/hotspot/default.nix:
```diff
@@ -64,6 +70,7 @@ mkDerivation rec {
 
   qtWrapperArgs = [
     "--prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ rustc-demangle ]}"
+    "--prefix PATH : ${lib.makeBinPath [ linuxPackages.perf ]}"
```
While we are at adding stuff to `PATH`, maybe we should add binutils too so that the disassembly view works?

K900 replied:
Wait it doesn't? I guess I tested it in a nix-shell...

alois31 replied:
nix-shell includes stdenv, which includes binutils, so it works there.

K900 commented:
Bunch of updates:
- removed the now-unnecessary .git hack
- pulled in patchelf-unstable to add rustc-demangle instead of LD_LIBRARY_PATH shenanigans
- added binutils
- moved PATH changes to go last instead of first, so user-added perf/binutils win

alois31 commented:
There is a very annoying regression that makes it inconvenient to use the "attach to process" or "profile system" modes, which I tracked down in https://github.com/KDAB/hotspot/pull/625. I recommend waiting for the situation to be resolved upstream and then pull an appropriate patch.

SuperSandro2000 reviewed (approved):
nix wise looks pretty solid to me
